qds_sdk/command_templates.py

```diff
@@ -0,0 +1,202 @@
+
+        #run command
+        run = subparsers.add_parser("run", help="Run a command template")
+        run.add_argument("id", help="Numeric id of the command template")
+        run.add_argument("--input_vars",
+            help="Add names and values for input variables", dest="input_vars", nargs="*")
+        run.set_defaults(func=CommandTemplateCmdLine.run_template)
+        CommandTemplateCmdLine.runparser = run
+
+        #run_and_wait command
+        run_and_wait = subparsers.add_parser("run_and_wait", 
+            help="Run a command template and wait for it to complete")
+
+        run_and_wait.add_argument("id", help="Numeric id of the command template")
+        run_and_wait.add_argument("--input_vars", 
+            help="Add name and values of input variables", dest="input_vars", nargs="*")
+
+        run_and_wait.set_defaults(func=CommandTemplateCmdLine.run_and_wait)
+        CommandTemplateCmdLine.run_and_waitparser = run_and_wait
+        return argparser
+
+    @staticmethod
+    def get_list_all_help():
+        return CommandTemplateCmdLine.list_allparser.format_help()
+
+    @staticmethod
+    def get_view_help():
+        return CommandTemplateCmdLine.viewparser.format_help()
+
+    @staticmethod
+    def get_remove_help():
+        return CommandTemplateCmdLine.removeparser.format_help()
+
+    @staticmethod
+    def get_run_help():
+        return CommandTemplateCmdLine.runparser.format_help()
+
+    @staticmethod
+    def get_run_and_wait_help():
+        return CommandTemplateCmdLine.run_and_waitparser.format_help()
+
+    @staticmethod
+    def run(args):
+        parser = CommandTemplateCmdLine.parsers()
+        parsed = parser.parse_args(args)
+        return parsed.func(parsed)
+
+    @staticmethod
+    def filter_fields(schedule, fields):
+        filtered = {}
+        try:
+            for field in fields:
+                filtered[field] = schedule[field]
+        except KeyError as e:
+            raise ParseError("Incorrect field name ",CommandTemplateCmdLine.get_list_all_help() )
+        return filtered
+
+    @staticmethod
+    def list_all(args):
+        commandtemplateslist = CommandTemplate.list_all(args)
+        if args.fields:
+            for s in commandtemplateslist:
+                s.attributes = CommandTemplateCmdLine.filter_fields(s.attributes, args.fields)
+        return json.dumps(commandtemplateslist, default=lambda o: o.attributes, sort_keys=True, indent=4)
+
+    @staticmethod
+    def view(args):
+        if args.id:
+            commandtemplate = CommandTemplate.find(args.id)
+        elif args.name:
+            commandtemplate = CommandTemplate.find_by_name(args.name)
+        else:
+            raise ParseError("Either template id or template name must be specified",CommandTemplateCmdLine.get_view_help() ) 
+        if args.fields:
+            commandtemplate.attributes = CommandTemplateCmdLine.filter_fields(schedule.attributes, args.fields)
```

Comment:
This would have been easily caught if you had added unit tests for all options. Please do that!